Implement `TokenCollector::visit(AST::FormatArgs&)`

Magnus-Mage opened the pull request:
Replace rust_sorry_at() stub with basic implementation that outputs format_args!() macro syntax.
Related to Issue - #4004 

This is WIP so let me know if this is ok for now or need changes.

CohenArthur commented on gcc/rust/expand/rust-macro-builtins-format-args.cc:
```diff
@@ -58,7 +58,7 @@ format_args_parse_expr (location_t invoc_locus, AST::MacroInvocData &invoc,
   // TODO(Arthur): Clean this up - if we haven't parsed a string literal but a
   // macro invocation, what do we do here? return a tl::unexpected?
-  rust_assert (format_expr->is_literal ());
+  // rust_assert (format_expr->is_literal ());
```
I don't think this needs to be part of this PR

Magnus-Mage commented:
Thank you for the suggestions.

I have made the necessary changes:

- Removed unnecessary `const_casts`, using const references where possible
- Reconstructed templates using `Fmt::ffi::Piece`, handling literals and placeholders
- Added support for format specifiers position, alignment, flags, width, precision, and type
- Argument handling for normal, named, and captured arguments

Also, tested this manually results [here](https://gist.github.com/877dd226bf40a6497bf9ee469d6a7c67.git)

Do let me know if the workaround is ok or need changes.
